tool/yarn

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+#
+# This script acts like the "yarn" command, but uses Tailscale's
+# currently-desired version, downloading it (and node) first if necessary.
+
+set -eu
+
+NODE_DIR="$HOME/.cache/tailscale-node"
+read -r YARN_REV < "$(dirname "$0")/yarn.rev"
+YARN_DIR="$HOME/.cache/tailscale-yarn"
+# This works for linux and darwin, which is sufficient
+# (we do not build for other targets).
+OS=$(uname -s | tr A-Z a-z)
+ARCH="$(uname -m)"
+if [ "$ARCH" = "aarch64" ]; then
+    # Go uses the name "arm64".
+    ARCH="arm64"
+elif [ "$ARCH" = "x86_64" ]; then
+    # Go uses the name "amd64".
+    ARCH="amd64"
+fi
+
+install_node() {
+    read -r REV < "$(dirname "$0")/node.rev"
+    NODE_URL="https://nodejs.org/dist/v${REV}/node-v${REV}-${OS}-${ARCH}.tar.gz"
+    install_tool "node" $REV $NODE_DIR $NODE_URL
+}
+
+install_yarn() {
+    YARN_URL="https://github.com/yarnpkg/yarn/releases/download/v$REV/yarn-v$REV.tar.gz"
+    install_tool "yarn" $YARN_REV $YARN_DIR $YARN_URL
+}
+
```

Comment:
The variables aren't correct, it didn't work on my machine until I made these changes.

```suggestion
install_node() {
    read -r NODE_REV < "$(dirname "$0")/node.rev"
    NODE_URL="https://nodejs.org/dist/v${NODE_REV}/node-v${NODE_REV}-${OS}-${ARCH}.tar.gz"
    install_tool "node" $NODE_REV $NODE_DIR $NODE_URL
}

install_yarn() {
    YARN_URL="https://github.com/yarnpkg/yarn/releases/download/v${YARN_REV}/yarn-v${YARN_REV}.tar.gz"
    install_tool "yarn" $YARN_REV $YARN_DIR $YARN_URL
}
```

Reply:
Oops, didn't do an end-to-end test. Only the `install_yarn` side is actually needed, fixed that.

Reply:
I only did the NODE_REV stuff to be consistent.

Reply:
Fair enough, and likely safer (I don't know, and don't want to learn bash/shell variable scoping/shadowing rules).

Reply:
Neither do I :)